web/src/app/core/services/token.service.ts

```diff
@@ -0,0 +1,106 @@
+import { Injectable } from '@angular/core';
+import { throwError, Observable } from 'rxjs';
+import { first, map, switchMap } from 'rxjs/operators';
+import { v4 as uuid } from 'uuid';
+
+// Dashboard model and services
+import { IProjectTokenModel, ProjectModel } from '../../shared/models/index.model';
+import { ProjectService } from './project.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TokenService {
+
+  constructor(
+    private projectService: ProjectService
+  ) { }
+
+  // This function returns the token details via id
+  public findOneById(projectUid: string, tokenUid: string): Observable<IProjectTokenModel> {
+    return this.projectService.findOneById(projectUid)
+      .pipe(
+        map((data: ProjectModel) => data && data.tokens && data.tokens.find((item: IProjectTokenModel) => item.uid === tokenUid))
+      );
+  }
+
+  // This function returns the tokens list
+  public findProjectTokens(projectUid: string): Observable<IProjectTokenModel[]> {
```

Comment:
Probably does not need `Tokens` in the method name as we are in the token service already.

May be `FindAllByProjectId`?

Reply:
agree, but may be `findAll`?

Reply:
Sounds good 👍